### ISSUE TITLE
How to add "app-name" attribute to scripts?

### ISSUE THREAD

**Issue by jpiquot** [oqtane/oqtane.framework#4058]
I am trying to create a Fluent UI theme for Oqtane.

I created a FluentUI module and added a Fluent UI theme inside.

How can I add the following scripts to Oqtane?
```html
<script app-name="{NAME OF YOUR APP}" src="./_content/Microsoft.FluentUI.AspNetCore.Components/js/initializersLoader.webview.js"></script>
<script src="_framework/blazor.webview.js"></script>
```
I don't see how to add the "app-name" attribute in the theme resources.
 
```csharp
public class ThemeInfo : ITheme
{
    public Theme Theme => new()
    {
        Name = "Hexalith Fluent UI",
        Version = "1.0.0",
        PackageName = "Hexalith.Module.MicrosoftFluentUI",
        ThemeSettingsType = "Hexalith.Themes.FluentUITheme.ThemeSettings, Hexalith.Module.MicrosoftFluentUI.Client.Oqtane",
        ContainerSettingsType = "Hexalith.Themes.FluentUITheme.ContainerSettings, Hexalith.Module.MicrosoftFluentUI.Client.Oqtane",
        Resources =
        [
            new Resource { ResourceType = ResourceType.Script, Url = "_content/Microsoft.FluentUI.AspNetCore.Components/js/initializersLoader.webview.js" },
            new Resource { ResourceType = ResourceType.Script, Url = "_framework/blazor.webview.js" },
        ]
    };
}

```

**Comment by leigh-pointer** [comment]
@jpiquot just a suggestion but cant you modify the script file and add your theme name, as that will be the the app-name?

**Comment by jpiquot** [comment]
@leigh-pointer the script is in Microsoft Fluent UI nuget packages. I can't change it.

**Comment by sbwalker** [comment]
"app-name" is not a standard HTML attribute... I am not sure why the Fluent UI team would have chosen such a strange requirement for registering their JavaScript. In addition, I am confused why you are specifying "_framework/blazor.webview.js" as a module resource... as it is Blazor framework library for .NET MAUI and would never be included in a module - it is already included here (Oqtane.Maui\wwwroot\index.html) and will be included when running Oqtane on .NET MAUI. Also the name of the Fluent UI library seems strange "initializersLoader.webview.js" - as this also seems to indicate that it is intended for .NET MAUI  (which relies on a WebView).

**Comment by jpiquot** [comment]
These are necessary to circumvent a MAUI bug, as detailed in the [Fluent UI for Blazor](https://www.fluentui-blazor.net/CodeSetup) documentation. Being new to Oqtane, my understanding of it is not yet comprehensive. Developing a module to incorporate Fluent UI serves as a proof of concept helps me to dig deeper in your code. 
I have another issue with the Microsoft.FluentUI.AspNetCore.Components package. It should automatically load all the required CSS and JS files when added to a project. However, this process doesn't seem to function with Oqtane. Is it necessary to manually include the files in the ITheme class to comply to it's dynamic module load?

**Comment by sbwalker** [comment]
So you are trying to use the Blazor Hybrid version of the Fluent UI components - which means you are building a .NET MAUI app and are trying to use Oqtane as a development framework for building components. Correct? 

Every UI component library has its own unique requirements for integration (and integration varies based on the version of Blazor/.NET you are targeting as well). In general, when using Oqtane there are 4 integrations:

1. Assemblies - need to exist in the /bin folder (including all dependencies)
2. CSS - need to be declared as Resources (which can be accomplished in Themes or Modules)
3. JavaScript - need to be declared as Resources (which can be accomplished in Themes or Modules). The way that JavaScript is executed in Blazor varies depending on render mode so you will need to ensure your approach aligns with the requirements of either Static or Interactive render mode.
4. Services - need to be registered using the appropriate Startup interface depending on whether the components execute on the client or server.

**Comment by jpiquot** [comment]
Thank you, @sbwalker, for your valuable information. Indeed, I am in the process of incorporating a module that adheres to the constraints of Blazor within MAUI. I have included a copy of the Fluent UI DLLs in the scripts and nuspec of the Package project. Found missing part within the placeholder.txt file located in the _content directory. It specifies that resources must be extracted from the Fluent UI NuGet packages and subsequently copied to the Module wwwroot/_content/PackageName/ folder.
